gen/gen.go

```diff
@@ -17,8 +17,27 @@ import (
 // Config exposes the options available for the code generation.
 type Config struct {
 	PackageName string
+	TagConfigs  []TagConfig
```

Comment:
Perhaps this could be simplified to a `map[string]TagConfig`? And maybe called `Tags`?

Reply:
`map[string]TagStyle` (`Style` instead of `Config`) maybe?